Frontend/ui-common/src/Engineering/Helpers/JBridge.js

```diff
@@ -1051,29 +1053,52 @@ export const requestAutoStartPermission = function (str) {
   };
 };
 
+export const animateCamera = function (lat) {
+  return function (lng) {
+    return function (zoom) {
+      return function (zoomType) {
+        return function () {
+          try {
+            window.JBridge.animateCamera(lat, lng, zoom, zoomType);
+          } catch (err) {
+            window.JBridge.animateCamera(lat, lng, zoom);
+          }
+        };
+      };
+    };
+  };
+};
+
 export const showMapImpl = function (id) {
   return function (isEnableCurrentLocation) {
     return function (type) {
       return function (zoom) {
-        return function (cb) {
-          return function (action) {
-            return function () {
-              const callback = callbackMapper.map(function (key, lat, lon) {
-                console.log("in show map", action);
-                window.x = cb;
-                window.y = action;
-                cb(action(key)(lat)(lon))();
-              });
-              const mapConfig = window && window.appConfig && window.appConfig.mapConfig ? JSON.stringify(window.appConfig.mapConfig) : "{}";
-              try {
-                window.JBridge.showMap(id, isEnableCurrentLocation, type, zoom, callback, mapConfig);
-              } catch (err) {
-                window.JBridge.showMap(id, isEnableCurrentLocation, type, zoom, callback);
-              }
-              return true;
+        return function(sourceLat){
+          return function(sourceLon){
+            return function (cb) {
+              return function (action) {
+                return function () {
+                  const callback = callbackMapper.map(function (key, lat, lon) {
+                    console.log("Lat long ::" , lat, lon);
+                    if(lat != 0.0 && lon != 0.0){
+                      animateCamera(sourceLat)(sourceLon)(zoom)("ZOOM")();
+                    }
```

Comment:
why we need this case here, its already handled in native, can you please mention In which cases its not working as expected